### PR TITLE
Prefer & as argSeparator and update comments

### LINKED ISSUE #13940: arg_separator breaks PMA: Call to a member function getCurrentUserAndHost
### Steps to reproduce
1. delete one row in "Browse"

### Expected behaviour
row get deleted, success

### Actual behaviour
```
Fatal error: Uncaught Error: Call to a member function getCurrentUserAndHost() on null in /var/www/htdocs/phpMyAdmin/libraries/check_user_privileges.inc.php:14
Stack trace:
#0 /var/www/htdocs/phpMyAdmin/sql.php(21): require_once()
#1 {main} thrown in /var/www/htdocs/phpMyAdmin/libraries/check_user_privileges.inc.php on line 14
```

### Server configuration
**Operating system**: Gentoo

**Web server:** lighttpd

**Database:** mariadb

**PHP version:** 7.1

**phpMyAdmin version:** git master rev 2823d27cd3

### Client configuration
**Browser:** Chrome

**Operating system:** Windows

it is triggered by arg_separator.input = "&;" php setting
PMA tends to use ";" which is ok most of the time, but for POST data, php (nor chrome F12 developer tools) don't recognize ; as separator, see php-src code at `http://git.php.net/?p=php-src.git;a=blob;f=main/php_variables.c;h=9e7f185bd944715b11d3e69005b533b99b983d65;hb=HEAD#l295`

the data being posted is:
`ajax_request=true&ajax_page_request=true&db=tracker;table=lkp;where_clause=%60lkp%60.%60userid%60+%3D+10+AND+%60lkp%60.%60type%60+%3D+%27gps%27;clause_is_unique=1;sql_query=SELECT+%2A+FROM+%60lkp%60;goto=sql.php;default_action=update;server=1&_nocache=1516126509642566829&token=.....`
as you can see it mix ; with & while both should be supported when in query string, it is not supported in post content, even Content-Type is `application/x-www-form-urlencoded; charset=UTF-8`

### LINKED ISSUE #14969: No Database Selected
1. Open SQL tab of a table
2. Execute default query
3. Execute again

pMA 4.8.6-dev

pma

### LINKED ISSUE #15004: phpMyAdmin No base has been selected
### Describe the bug

I am using phpMyAdmin via Xampp latest version under W10 build 1809.

All that goes on normally and when I display a table of any database to continue in the pages I get the message:

`1046 - No base has been selected`
While we can see in the following picture that I am well in the right base selected

The message arrives once I use the command > or >> (in yellow) once the table is already displayed.

Here I need your lights to understand what is happening. 
Thanks in advance. 
Alain

### To Reproduce

3. Scroll down to once the table is already displayed.
4. 1046 - No base has been selected

### Screenshots
phpmyadmin

### Server configuration

- Operating system: W10 build 1809
- Web server:
- Database version:
- PHP version: 7.3.2
- phpMyAdmin version: 4.8.5

### Client configuration

- Browser: chrome last version
- Operating system: W10

### LINKED ISSUE #16826: POST & arg_separator=";"
PMA forge POST data (to jQuery) with `;` used as arg_separator instead of `&`!

A good practice is to **never** use a `string` in options.data (jQuery) when you don't known options.type (GET or POST?).

### PHP Config

arg_separator.output=";"
arg_separator.input=";&"

### To Reproduce

Edit a row from any table/db (click any pencil), throw the `No databases selected.` error from librairies/db_table_exists.inc.php:34.

### To Trace bug + Ugly Patch

Edit js/functions.js and add a few lines in the $.ajaxPrefilter

```js
$.ajaxPrefilter(function (options, originalOptions, jqXHR) {
    var nocache = new Date().getTime() + '' + Math.floor(Math.random() * 1000000);
    if (typeof options.data === 'string') {
        // PATCH START HERE
        if (options.type.toUpperCase() == "POST" && options.data.indexOf(";") >= 0) {
            console.log("POST ';' ERROR: " + options.data); // Trace bug
            options.data = options.data.replace(/;/g, "&"); // Ugly Patch
        }
        // PATCH END HERE
        options.data += '&_nocache=' + nocache + '&token=' + encodeURIComponent(PMA_commonParams.get('token'));
    } else if (typeof options.data === 'object') {
        options.data = $.extend(originalOptions.data, { '_nocache' : nocache, 'token': PMA_commonParams.get('token') });
    }
});
```

### Config (irrelevant because PMA should follow standard anyway!)

Ubuntu 20.04.2 LTS
PMA version : 4.9.5deb2
PHP Version : 7.4.3
MariaDB

See also: https://github.com/phpmyadmin/phpmyadmin/issues/13940 (a closed issue with same problem in older version of PMA)

### PR DESCRIPTION
**Pull request by Melnofil**
### Description

PHPMyAdmin should prefer `$argSeparator = '&';` until is fully compatible with a custom argument separator.

- Fixes #13940
- Fixes #14969
- Fixes #15004
- Fixes #16826

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: libraries/classes/Url.php [hunk 1]
```diff
@@ -308,25 +308,25 @@ public static function decryptQuery(string $query): string|null
...
 
         if ($separator === null) {
-            // use separators defined by php, but prefer ';'
+            // use separators defined by php, but prefer '&', or ';'
             // as recommended by W3C
             // (see https://www.w3.org/TR/1999/REC-html401-19991224/appendix
             // /notes.html#h-B.2.2)
             $argSeparator = (string) ini_get('arg_separator.input');
-            if (str_contains($argSeparator, ';')) {
+            if (str_contains($argSeparator, '&') || $argSeparator === '') {
```
Comment (williamdes) [inline libraries/classes/Url.php:324]:
```suggestion
            if ($argSeparator === '&' || $argSeparator === '') {
```

File: libraries/classes/Url.php [hunk 2]
```diff
@@ -308,25 +308,25 @@ public static function decryptQuery(string $query): string|null
...
+                $separator = '&';
+            } elseif (str_contains($argSeparator, ';')) {
```
Comment (williamdes) [inline libraries/classes/Url.php:326]:
```suggestion
            } elseif ($argSeparator === ';') {
```

File: libraries/classes/Url.php (hunk 1 above)
Comment (Melnofil) [reply libraries/classes/Url.php:324]: Cette suggestion ne peut pas être acceptée, car le test === '&' ne fonctionnera pas si en configuration, on accepte plusieurs caractères (par ex. `;&`).

File: libraries/classes/Url.php (hunk 2 above)
Comment (Melnofil) [reply libraries/classes/Url.php:326]: Cette suggestion ne peut pas être acceptée, car le test === ';' ne fonctionnera pas si en configuration, on accepte plusieurs caractères (par ex. `/&;`).

**Review by MauricioFauth — APPROVED** [review]

**Comment by MauricioFauth** [comment]
Hello. Thanks for your contribution!

Could you add a Signed-off-by line to your commit with your full name?
For example: `Signed-off-by: Jane Developer <jane@example.org>`

You can also add a comment to this pull request with your Signed-off-by line.

See [Developer's Certificate of Origin](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO#L39)

**Pull request closed without merging**